daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -886,20 +886,20 @@ func assertSimulationResultsEqual(t *testing.T, expectedError string, expected,
 	require.Equal(t, expected, actual)
 }
 
-func makePendingTxnResponse(t *testing.T, txn transactions.SignedTxnWithAD, handle codec.Handle) model.PendingTransactionResponse {
+func makePendingTxnResponse(t *testing.T, txn transactions.SignedTxnWithAD) v2.PreEncodedTxInfo {
 	t.Helper()
 	preEncoded := v2.ConvertInnerTxn(&txn)
 
-	// encode to bytes
-	var encodedBytes []byte
-	encoder := codec.NewEncoderBytes(&encodedBytes, handle)
-	err := encoder.Encode(&preEncoded)
-	require.NoError(t, err)
+	// In theory we could return preEncoded directly, but there appears to be some subtle differences
+	// once you encode and decode the object, such as *uint64 fields turning from 0 to nil. So to be
+	// safe, let's encode and decode the object.
+
+	// Encode to bytes
+	encodedBytes := protocol.EncodeReflect(&preEncoded)
 
-	// decode to model.PendingTransactionResponse
-	var response model.PendingTransactionResponse
-	decoder := codec.NewDecoderBytes(encodedBytes, handle)
-	err = decoder.Decode(&response)
+	// Decode to v2.PreEncodedTxInfo
+	var response v2.PreEncodedTxInfo
+	err := protocol.DecodeReflect(encodedBytes, &response)
```

Comment:
Yes, I believe `omitempty` is causing `uint64` pointers to 0 to be decoded as `nil`. Rather than trying to manually reproduce this change on `preEncoded`, I thought it would be simpler to just encoded and decode that expected object before comparison.

Reply:
sounds good, that makes sense.